e2e/donation-modal.spec.ts

```diff
@@ -34,123 +35,86 @@ const completeFrontEndCert = async (page: Page) => {
-      solution: `// Setup\nvar a;\n\n// Only change code below this line\na = 7;`
-    },
-    {
-      url: '/learn/javascript-algorithms-and-data-structures/basic-javascript/assigning-the-value-of-one-variable-to-another',
-      solution: `// Setup\nvar a;\na = 7;\nvar b;\n\n// Only change code below this line\nb = a;`
-    },
-    {
-      url: '/learn/javascript-algorithms-and-data-structures/basic-javascript/initializing-variables-with-the-assignment-operator',
-      solution: 'var a = 9;'
-    },
-    {
-      url: '/learn/javascript-algorithms-and-data-structures/basic-javascript/declare-string-variables',
-      solution: `var myFirstName = 'foo';\nvar myLastName = 'bar';`
-    },
-    {
-      url: '/learn/javascript-algorithms-and-data-structures/basic-javascript/understanding-uninitialized-variables',
-      solution: `// Only change code below this line\nvar a = 5;\nvar b = 10;\nvar c = 'I am a';\n// Only change code above this line\n\na = a + 1;\nb = b + 5;\nc = c + " String!";`
-    },
-    {
-      url: '/learn/javascript-algorithms-and-data-structures/basic-javascript/understanding-case-sensitivity-in-variables',
-      solution: `// Variable declarations\nvar studlyCapVar;\nvar properCamelCase;\nvar titleCaseOver;\n\n// Variable assignments\nstudlyCapVar = 10;\nproperCamelCase = "A String";\ntitleCaseOver = 9000;`
-    },
-    {
-      url: '/learn/javascript-algorithms-and-data-structures/basic-javascript/explore-differences-between-the-var-and-let-keywords',
-      solution: `let catName = "Oliver";\nlet catSound = "Meow!";`
-    },
-    {
-      url: '/learn/javascript-algorithms-and-data-structures/basic-javascript/declare-a-read-only-variable-with-the-const-keyword',
-      solution: `const FCC = "freeCodeCamp";\n// Change this line\nlet fact = "is cool!";\n// Change this line\nfact = "is awesome!";\nconsole.log(FCC, fact);\n// Change this line`
-    }
-  ];
-
-  for (const challenge of challenges) {
+  await page.goto(challenges[0].url);
+  for (const challenge of challenges.slice(0, number)) {
     await page.waitForURL(challenge.url);
-
     await focusEditor({ page, isMobile });
     await clearEditor({ page, browserName });
-
     await page.evaluate(
       async contents => await navigator.clipboard.writeText(contents),
       challenge.solution
     );
     await page.keyboard.press('ControlOrMeta+V');
-
     await page.getByRole('button', { name: 'Run' }).click();
-    await expect(page.getByRole('dialog')).toBeVisible(); // completion dialog
+    await expect(
+      page.getByRole('dialog').filter({ hasText: 'Basic Javascript' })
+    ).toBeVisible(); // completion dialog
     await page.getByRole('button', { name: 'Submit' }).click();
+    if (checkModal) {
+      const donationModal = page
+        .getByRole('dialog')
+        .filter({ hasText: 'Become a Supporter' });
+      if (await donationModal.isVisible()) {
+        await expect(donationModal).toBeVisible();
+        return;
+      }
```

Comment:
The test It looks for the modal in each submission, and only completes the test when/if it finds one.

That is why the suggestion would fail since the donation modal would not exist on the first submission:

Reply:
Right, but I think you want to test if the donation modal is ever shown and fail if it doesn't. This test doesn't do that.

In fact the test cannot fail:

- If modal is not visible, skip assertion -> pass
- If modal IS visible, check if visible (it is!) -> pass

What I think you want is to have a variable `isDonationShown`, initially set to false, set it to true if the modal pops up and assert that it's true at the end.

Reply:
I see, I changed the test accordingly.